### DISCUSSION TITLE
When the Parameters of Tuning allocations are configured, then there is no dynamic memory in DDS working completely ？

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3685] [General]
Is it right？Is there any exception?
Just follow the example: https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/realtime/allocations.html

**Comment by JesusPoderoso** [comment]
The only exception is if the allocation parameter is configured as zero. 
A value of zero implies no size limitation, and memory will be dynamically allocated as needed. By default, a full dynamic behavior is used.

**Reply by JesusPoderoso** [reply]
Note: That applies only memory management operations in the main loop, as decribed in the [header of the documentation section](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/realtime/allocations.html#tuning-allocations).
During initialization, the application may perform memory allocation dynamically to reach the allocated resources set as parameters.

**Reply by wjhtinger** [reply]
The memory allocation in initialization only run once in process, right?

**Reply by JesusPoderoso** [reply]
Yes, you are right
